src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -357,7 +358,7 @@ private List<Target> ReadTargets(ref Utf8JsonReader reader)
 
             while (reader.Read() && reader.IsTokenTypeProperty())
             {
-                targets.Add(ReadTarget(ref reader, reader.GetString()));
+                targets.Add(ReadTarget(ref reader, reader.GetString()!));
```

Comment:
I think this `!` is lying that this can't be `null`. We should make the `ReadTarget` method accept `string? targetName`.

Reply:
Similar comment for `ReadTargetLibrary`

Reply:
Experimented with it and looks like `targetName` shouldn't be null, or code will NRE elsewhere:

- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs#L195 (`IsRuntimeTarget` would crash if `Name` is null)
- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs#L202 (`new TargetInfo(framework...)` a bit later would crash if `framework` is null)
- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs#L238  (`IsRuntimeTarget` would crash if `Name` is null)

Reply:
I've added `Debug.Assert` here just in case